docs/source/csdl-spec.md

```diff
@@ -0,0 +1,630 @@
+---
+title: CSDL Specification
+sidebar_title: CSDL Specification
+description: The Composed Schema Definition Language
+status: Draft
+version: "1.0.0"
+authors: 
+  - Jake Dawkins <jake@apollographql.com>
+  - Trevor Scheer <trevor@apollographql.com>
+  - Ashi Krishnan <ashi@apollographql.com>
+---
+
+## Introduction: What is CSDL?
+
+The Composed Schema Definition Language (CSDL) provides a way to describe a GraphQL Schema which is composed from one or more other GraphQL schemas.
+
+The CSDL is intended to be a single artifact which can power a graph router. For example, the CSDL replaces the service list configuration of Apollo Gateway.
```

Comment:
nitpick: As a reader, this feels like a detail that I could take in two ways. One of those is, I think, intended -> "Currently, I see the configuration to my gateway as `serviceList` and now I should see configuration to my gateway as CSDL". The other (and where my head first went) is confusing -> "Currently, the `serviceList` tells my gateway how to compose. Now CSDL will replace that".

I think that, even more concretely, we may very well have an intermediate period where the `serviceList` configuration of Apollo gateway leads to a new version of composition running at startup of that server, and thus producing CSDL for the Apollo gateway to use as configuration of its query planner.

All to say, I think that maybe there's a small disagreement — I think the CSDL is intended to be a single artifact which can power a query planner. A graph router (aside: currently we don't call Apollo gateway that publicly; should we introduce that term here?) may contain more things as configuration, such as plugins to lifecycle, and it may even take an input to _create_ or _fetch_ a CSDL to pass to its component, a query planner.